android/src/main/java/ee/forgr/audio/NativeAudio.java

```diff
@@ -255,6 +283,276 @@ public void run() {
+                            return;
+                        }
+
+                        boolean autoPlay = call.getBoolean("autoPlay", true);
+                        final boolean deleteAfterPlay = call.getBoolean("deleteAfterPlay", false);
+                        float volume = call.getFloat(VOLUME, 1F);
+                        boolean isLocalUrl = call.getBoolean("isUrl", false);
+                        int audioChannelNum = 1; // Single channel for playOnce
+
+                        // Track this as a playOnce asset
+                        plugin.playOnceAssets.add(assetId);
+
+                        // Store notification metadata if provided
+                        JSObject metadata = call.getObject(NOTIFICATION_METADATA);
+                        if (metadata != null) {
+                            Map<String, String> metadataMap = new HashMap<>();
+                            if (metadata.has("title")) metadataMap.put("title", metadata.getString("title"));
+                            if (metadata.has("artist")) metadataMap.put("artist", metadata.getString("artist"));
+                            if (metadata.has("album")) metadataMap.put("album", metadata.getString("album"));
+                            if (metadata.has("artworkUrl")) metadataMap.put("artworkUrl", metadata.getString("artworkUrl"));
+                            if (!metadataMap.isEmpty()) {
+                                plugin.notificationMetadataMap.put(assetId, metadataMap);
+                            }
+                        }
+
+                        // Preload the asset using the helper method
+                        try {
+                            // Check if asset already exists
+                            if (plugin.audioAssetList.containsKey(assetId)) {
+                                throw new Exception(ERROR_AUDIO_EXISTS + " - " + assetId);
+                            }
+
+                            // Load the asset using the helper method
+                            JSObject headersObj = call.getObject("headers");
+                            AudioAsset asset = plugin.loadAudioAsset(assetId, assetPath, isLocalUrl, volume, audioChannelNum, headersObj);
+
+                            if (asset == null) {
+                                throw new Exception("Failed to load asset");
+                            }
+
+                            // Add to asset list; completion listener is set below with cleanup
+                            plugin.audioAssetList.put(assetId, asset);
+
+                            // Store the file path if we need to delete it later
+                            final String filePathToDelete;
+                            if (deleteAfterPlay && isLocalUrl && assetPath.startsWith("file://")) {
+                                filePathToDelete = assetPath;
+                            } else {
+                                filePathToDelete = null;
+                            }
```

Comment:
Java supports ternary operator. You can rewrite this as:

```suggestion
final String filePathToDelete =
        (deleteAfterPlay && isLocalUrl && assetPath.startsWith("file://"))
        ? assetPath
        : null;
```